RevenueCat/Plugins/iOS/PurchasesUnityHelper.m

```diff
@@ -265,7 +265,15 @@ - (void)promotionalOfferForProductIdentifier:(NSString *)productIdentifier
     [RCCommonFunctionality promotionalOfferForProductIdentifier:productIdentifier
                                                       discount:discountIdentifier
                                                completionBlock:^(NSDictionary *_Nullable responseDictionary, RCErrorContainer *_Nullable error) {
-        [self sendJSONObject:responseDictionary toMethod:GET_PROMOTIONAL_OFFER];
+        NSMutableDictionary *response;
+        if (error) {
+            response = [NSMutableDictionary new];
+            response[@"error"] = error.info;
+
+        } else {
+            response = [NSMutableDictionary dictionaryWithDictionary:responseDictionary];
+        }
+        [self sendJSONObject:response toMethod:GET_PROMOTIONAL_OFFER];
```

Comment:
Maybe send an immutable copy?
```suggestion
        [self sendJSONObject:response.copy toMethod:GET_PROMOTIONAL_OFFER];
```

Reply:
Actually I think you could simplify this while thing like this:
```objc
NSDictionary *response = (error)
? @{
  @"error": error.info
}
: responseDictionary;
```

Reply:
I don't understand this language

Reply:
Hmm right, that seems to be the same it is right now (with different indentation)?